tests/unit/igraph_circulant.c

```diff
@@ -0,0 +1,176 @@
+/*
+   IGraph library.
+   Copyright (C) 2021  The igraph development team <igraph@igraph.org>
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include <igraph.h>
+
+#include "test_utilities.inc"
+
+int main() {
+
+    igraph_t graph, graph_test;
+    igraph_vector_int_t l;
+    igraph_bool_t iso;
+    igraph_integer_t i;
+
+    /* Testing n = 1 case */
+    /* n = 1, l = [1], undirected */
+    igraph_vector_int_init(&l, 0);
+    igraph_vector_int_push_back(&l, 1);
+    IGRAPH_ASSERT(igraph_circulant(&graph, 1, &l, IGRAPH_UNDIRECTED) == IGRAPH_SUCCESS);
+    igraph_small(&graph_test, 1, IGRAPH_UNDIRECTED, -1);
+    IGRAPH_ASSERT(igraph_is_same_graph(&graph, &graph_test, &iso) == IGRAPH_SUCCESS);
```

Comment:
This technique is okay, but for the future, the alternative is simply to print the graph to `stdout` with `print_graph_canon()` and provide an expected output file in `igraph_circulant.out`. Our unit test framework will compare the expected output with the printed output and print a diff if they are different so it's easier to see which edges are missing or which edges have been added incorrectly.

Reply:
Yep that would have been alot more natural. I'll keep it in mind for next time!